src/js/MainRouter.jsx

```diff
@@ -19,18 +19,17 @@ const TRANSLATION_PREFIXES = ['default', 'dashboard', 'combinedShipments', 'prod
 
 class MainRouter extends React.Component {
   componentDidMount() {
-    this.props.fetchSessionInfo().then(() => {
-      this.props.initialize({
-        languages: this.props.supportedLocales,
-        options: {
-          renderToStaticMarkup,
-          onMissingTranslation,
-        },
-      });
-      this.props.setActiveLanguage(this.props.locale);
-      this.props.fetchMenuConfig();
-      TRANSLATION_PREFIXES.forEach(prefix => this.props.fetchTranslations('', prefix));
+    this.props.fetchSessionInfo();
+    this.props.initialize({
```

Comment:
On the initial load `this.props.supportedLocales` and `this.props.locale` will be empty and because you are not waiting for the data from `this.props.fetchSessionInfo()` you might wrongly initialize translations.

Reply:
I probably forgot to change it back, as I was trying to find any possible solution for my issue. Will bring it back